src/modules/security.js

```diff
@@ -512,11 +509,82 @@
       errorMessageKey: 'requiredFields'
     });
 
+    /*
+     * Google reCaptcha 2
+     */
+    $.formUtils.addValidator({
+        name: 'recaptcha',
+        validatorFunction: function (val, $el)
+        {
+            return grecaptcha.getResponse($el.data('validation-recaptcha-widget-id'));
+        },
+        errorMessage: '',
+        errorMessageKey: 'badreCaptcha'
+    });
 
-
-  $.fn.displayPasswordStrength = function(conf) {
+    $.fn.displayPasswordStrength = function(conf) {
         new $.formUtils.validators.validate_strength.strengthDisplay(this, conf);
         return this;
     };
 
+    var setupGooglereCaptcha = function (evt, $forms, config)
+    {
+        if( typeof grecaptcha !== typeof undefined ){
+            throw new Error('reCaptcha API can not be loaded by hand, delete reCaptcha API snippet.');
+        }
+
```

Comment:
Actually it is necessary. We need to make sure that callback function `reCaptchaLoaded` will be called and if we let user to load API we can't be sure that he actually used good callback function name and set `render=explicit`. 

Actually I didn't checked if `modulesLoaded` and `formValidationSetup` can be called several times - my bad. Fixed it like you suggested.